<launch>
    <arg name="launch_prefix" default="" />
    <arg name="node_namespace" default="apriltag_ros_continuous_node" />
    <arg name="camera_name" default="/xtion/rgb" />
    <arg name="image_topic" default="image_raw" />
    <arg name="tag_location_file" default="tag_real_world.yaml"/>

    <arg name="localization_method" default="weighted_average"/>
    <arg name="buffer_size" default="20"/>
    <arg name="max_time_diff" default="1.3"/>
    <arg name="max_error" default="0.25"/>

    <arg name="save_location" default="$(find apriltag_localization)/real_tests/data"/>
    <arg name="save_name" default="test_buf_er_tdiff"/>
    <arg name="save_format" default=".csv"/>
    <arg name="test_type" default="straight"/>


    <rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml" ns="$(arg node_namespace)" />
    <rosparam command="load" file="$(find apriltag_ros)/config/real_tags.yaml" ns="$(arg node_namespace)" />
    
    <rosparam command="load" file="$(find apriltag_localization)/config/$(arg tag_location_file)" ns="apriltag_localization" />
    
    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
        <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
        <remap from="camera_info" to="$(arg camera_name)/camera_info" />
        <param name="publish_tag_detections_image" type="bool" value="true" /> 
    </node>

    <node pkg="apriltag_localization" name="tag_localization_node" type="real_world_detection_node.py" output="screen">
        <param name="localization_method" type="string" value="$(arg localization_method)" />
        <param name="buffer_size" type="int" value="$(arg buffer_size)" />
        <param name="max_time_diff" type="double" value="$(arg max_time_diff)" />
        <param name="max_error" type="double" value="$(arg max_error)" />
    </node>

    <node pkg="apriltag_localization" name="real_world_test" type="real_world_final_test.py" output="screen">
        <param name="save_location" type="string" value="$(arg save_location)"/>
        <param name="save_name" type="string" value="$(arg save_name)"/>
        <param name="save_format" type="string" value="$(arg save_format)"/>
        <param name="test_type" type="string" value="$(arg test_type)"/>
    </node>
</launch>